Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/security/DccTicketingSecurityTool.kt

```diff
@@ -0,0 +1,87 @@
+package de.rki.coronawarnapp.dccticketing.core.security
+
+import de.rki.coronawarnapp.dccticketing.core.common.DccTicketingException
+import de.rki.coronawarnapp.util.encoding.base64
+import de.rki.coronawarnapp.util.encryption.rsa.RSACryptography
+import de.rki.coronawarnapp.util.security.Sha256Signature
+import java.security.PrivateKey
+import java.security.PublicKey
+import javax.inject.Inject
+
+class DccTicketingSecurityTool @Inject constructor(
+    private val dccTicketingCryptography: DccTicketingCryptography,
+    private val rsaCryptography: RSACryptography,
+    private val sha256Signature: Sha256Signature
+) {
+
+    fun encryptAndSign(input: Input): Output {
+        val key = dccTicketingCryptography.generateSecureRandomKey()
+        val encryptedDCC = input.encryptDcc(key)
+        return Output(
+            encryptedDCCBase64 = encryptedDCC.base64(),
+            encryptionKeyBase64 = input.encryptKey(key),
+            signatureBase64 = encryptedDCC.signWith(input.privateKeyForSigning),
+            signatureAlgorithm = Sha256Signature.ALGORITHM
+        )
+    }
+
+    private fun Input.encryptDcc(key: ByteArray): EncryptedDcc = try {
+        when (encryptionScheme) {
+            Scheme.RSAOAEPWithSHA256AESCBC -> dccTicketingCryptography.encryptWithCBC(
+                iv = nonceBase64,
+                data = dccBarcodeData,
+                key = key
+            )
+            Scheme.RSAOAEPWithSHA256AESGCM -> dccTicketingCryptography.encryptWithGCM(
+                iv = nonceBase64,
+                data = dccBarcodeData,
+                key = key
+            )
+        }
+    } catch (e: DccTicketingException) {
+        throw e
+    } catch (e: Exception) {
+        // anything else
+        throw DccTicketingException(DccTicketingException.ErrorCode.RSA_ENC_NOT_SUPPORTED)
```

Comment:
`encryptWithCBC()` and `encryptWithGCM()` throw only `DccTicketingException` 
shouldn't `RSA_ENC_NOT_SUPPORTED` be thrown by `input.encryptKey(key)` at least it mentioned like this in the spec https://github.com/corona-warn-app/cwa-app-tech-spec/blob/0f1e3a323fd2d52917ffc71ea256d26278e4e16b/docs/spec/dcc-validation-service.md#encrypting-data-with-rsa-oaep-with-sha256

Reply:
True, that exception needs to be moved, thanks